File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node


class MyNode(Node):

    def __init__(self):
        super().__init__("py_test")

        # add counter
        self.counter = 0

        # Use logger to print text
        self.get_logger().info("Hello ROS2!!!")

        # create a timer to execute periodically
        self.create_timer(0.5, self.timer_callback)

    def timer_callback(self):
        self.counter += 1
        self.get_logger().info("Hello " + str(self.counter))


def main(args=None):

    # Start the ros communication
    rclpy.init(args=args)

    # Create a node
    node = MyNode()

    # will paused the program and continue to be alive
    rclpy.spin(node)

    # Last sentence to suspend communication
    rclpy.shutdown()


if __name__=="__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"


class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("cpp_test"), counter_(0)
    {
        // Print log
        RCLCPP_INFO(this->get_logger(),"Hello CPP Node");

        timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                         std::bind(&MyNode::timerCallback, this));
        
    }

private:

    void timerCallback()
    {
        counter_++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};



int main(int argc, char **argv)
{
    // Initialize ROS2 communication
    rclcpp::init(argc, argv);

    // Create a pointer for a node obj
    auto node = std::make_shared<MyNode>();

    // Pause the program and keep the node alive
    rclcpp::spin(node);

    // Node will be shutdown
    rclcpp::shutdown();

    return 0;
}